File: Ciklo/src/main/java/com/example/ciklo/model/Bill.java
package com.example.ciklo.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.Nationalized;

import java.util.UUID;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Entity
@Table(name = "_bill")
public class Bill {
    @Id
//    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private UUID bid;
    @ManyToOne(fetch = FetchType.EAGER)
    @ToString.Exclude
    private Customer cus;
    @ManyToOne(fetch = FetchType.EAGER)
    @ToString.Exclude
    private Driver driver;
    @ManyToOne(fetch = FetchType.EAGER)
    @ToString.Exclude
    private Cyclo cyclo;
    @Nationalized
    private String start_loc;
    @Nationalized
    private String end_loc;
    private String distance;
    private String dateTrip;
    private double total;

    public Bill(Customer cus, Driver driver, Cyclo cyclo, String startLoc, String endLoc, String distance, String time, double total) {
        this.cus = cus;
        this.driver = driver;
        this.cyclo= cyclo;
        this.start_loc = startLoc;
        this.end_loc=endLoc;
        this.distance=distance;
        this.dateTrip = time;
        this.total = total;
    }
}

File: Ciklo/src/main/resources/static/custom/js/bill.js
// view detail bill
function viewAllBills() {
    var tb = $('#example').DataTable()
    $.ajax(
        {
            url: "/ciklo/bill/viewBill",
            type: 'GET',
            data:
                {
                },
            error: function () {
                console.log("error")
            },
            success: function (data) {
                console.log(data);
                if (data != null) {
                    appendToTable(data, tb);
                }
            }
        }
    )
}

//append data to table for driver and customer
function appendToTable(data, tb) {
    tb.clear().draw()
    for (let i = 0; i < data.length; i++) {
        tb.row.add(
            [
                data[i].bid,
                (data[i].driver.dfirstname + " " + data[i].driver.dlastname),
                data[i].cyclo.reg_No,
                data[i].start_loc,
                data[i].end_loc,
                data[i].distance,
                data[i].dateTrip,
                data[i].total
            ]).draw(false);
    }
}



